plugin/db/spanner/sync.go

```diff
@@ -24,7 +26,23 @@ func (d *Driver) SyncInstance(ctx context.Context) (*db.InstanceMeta, error) {
 		if err != nil {
 			return nil, err
 		}
-		databaseList = append(databaseList, db.DatabaseMeta{Name: database.Name})
+
+		match := dsnRegExp.FindStringSubmatch(database.Name)
```

Comment:
Add unit tests and some comments on why we need to do this.

Reply:
done